src/hotspot/share/oops/method.cpp

```diff
@@ -335,7 +335,8 @@ int Method::bci_from(address bcp) const {
 
 
 int Method::validate_bci(int bci) const {
-  return (bci == 0 || bci < code_size()) ? bci : -1;
+  // Called from the verifier, and should return -1 if not valid.
+  return ((is_native() && bci == 0) || (!is_native() && 0 <= bci && bci < code_size())) ? bci : -1;
```

Comment:
This expansion seems more correct, but it also seems unrelated to the main changes.

Reply:
This expansion was a bug that I hit while testing the main change.

Reply:
I removed all the code for the VerifyError <init> code to reduce the test case, so the failing BCI was 0 and the verifier reporting tested whether this was a valid bytecode, then asserted a couple lines further that it wasn't a valid bytecode.